DS-3112 Default dc.identifier.uri config shouldn't assume handle serv…

bram-atmire opened the pull request:
…er will be used

https://jira.duraspace.org/browse/DS-3112

bram-atmire committed:
DS-3112 Default dc.identifier.uri config shouldn't assume handle server will be used

tdonohue commented on local.cfg.EXAMPLE:
```diff
@@ -145,26 +145,26 @@ db.schema = public
-# (Defaults to a dummy/fake prefix of 123456789)
-#handle.prefix = 123456789
+# After registering with handle.net, enter the prefix assigned to your organization here.
+handle.prefix = internal
```
I like this PR overall. One tiny thing to make others aware of. This *does* change the default handle prefix from "123456789" to "internal".  I think that seems ok (as long as the rest of our code respects a non-number based prefix, and I think it should).  But, that may be worth documenting explicitly in our Release Notes (in case future people wonder when this switch of default prefixes came about).

mwoodiupui replied:
Is there a need to replace the well-known Handle equivalent of "example.com"?  That prefix is defined as belonging to no one, for testing and demonstration, IIRC.

tdonohue replied:
@bram-atmire : one thing I just realized.  The configs in `local.cfg.EXAMPLE` are commented out on purpose (because the defaults all reside in dspace.cfg, and *everything* is optional in `local.cfg`).  So, this PR likely needs also change default settings in dspace.cfg.  Otherwise, if a config isn't specified in your local.cfg, the existing defaults in dspace.cfg will be in affect.

bram-atmire replied:
@mwoodiupui "well known" lies in the eye of the beholder I suppose. To people who are unfamiliar with this, 123456789 looks unnecessary long and "generic" to be part of the urls. The proposed alternative "internal" can make it easier to make the distinction between internally assigned urls of DSpace, and permalinks/persistent urls like an official handle or DOI.

tdonohue replied:
Regarding the "well known" comment, we *would* want to ensure this change from `123456789` -> `internal` is well documented.  My impression from Friedrich Summann from BASE (who reported this issue in a DCAT meeting) was that he actually started to make his own code-level assumptions when he encountered handles starting with `123456789`.  So, we'd need to let him (and others who make similar assumptions) know that the default has been *switched* to `internal`.

bram-atmire committed:
DS-3112 setting handle defaults in dspace.cfg

following Tim's suggestion, also changing this in dspace.cfg instead of only in local.cfg.EXAMPLE

bram-atmire commented:
Good catch @tdonohue - it indeed makes more sense to change it directly in dspace.cfg as well if we're talking about default.

About the concern of a non-numerical default: we have clients running this in production for a while, for example, https://rdmc.nottingham.ac.uk/handle/internal/43

tdonohue commented on local.cfg.EXAMPLE:
```diff
@@ -145,26 +145,26 @@ db.schema = public
 #
 # If omitted, the canonical URL prefix will be http://hdl.handle.net/
-#handle.canonical.prefix = http://hdl.handle.net/
+handle.canonical.prefix = ${dspace.url}/handle/
```
I'd recommend re-commenting out this `handle.canonical.prefix` example in `local.cfg.EXAMPLE`, along with the `handle.prefix` below.  Now that these defaults have been moved to `dspace.cfg`, there is no need for the example `local.cfg` to override them by default.

tdonohue commented:
After further investigation, we use a dummy prefix of 123456789 in many areas of the codebase.
https://github.com/DSpace/DSpace/search?p=2&q=123456789&utf8=%E2%9C%93

We might want to split this PR in two.  One that simply fixes the core issue behind DS-3112, and a second that replaces "123456789" with "internal" (as the latter is a likely bigger PR)

mwoodiupui commented:
If we want to change the dummy prefix, we should manifest it somewhere and use the symbol instead.

tdonohue commented:
Closing this PR until we decide whether to change the default prefix to "internal" (see latest comments on tickets). 

The changing of the default setting for `handle.canonical.prefix` was already accepted/merged in #1535
